cmd/bedValueWig/bedValueWig.go

```diff
@@ -27,54 +32,68 @@ func bedMaxWig(infile string, database string, chromsizeFile string, outfile str
 			var wigCounterByChrom float64 = 0
 			for k := range wigData[i].Values { //Cycle through each value in the float64[]
 				var chromValueMultiplyByStep float64 = 0
-				chromValueMultiplyByStep = (float64(wigData[i].Step) * wigData[i].Values[k]) // multiply each value by the step for that chrom
+				chromValueMultiplyByStep = float64(wigData[i].Step) * wigData[i].Values[k] // multiply each value by the step for that chrom
 				wigCounterByChrom = wigCounterByChrom + chromValueMultiplyByStep
 			}
 			wigTotal += wigCounterByChrom
 		}
 	}
 
 	for i = 0; i < len(sizes); i++ {
-		chromSlice = WigChromToSlice(wigData, sizes[i].Size, sizes[i].Name)
+		chromIndex = getWigIndex(wigData, sizes[i].Name)
 		for k := 0; k < len(records); k++ {
 			if records[k].Chrom == sizes[i].Name {
 				currentBed = records[k]
 				if currentBed.FieldsInitialized < 7 {
 					currentBed.FieldsInitialized = 7
 				}
-				maxWig := bedRangeMax(chromSlice, records[k].ChromStart, records[k].ChromEnd)
+				if bedMinWigFlag {
+					currValue = bedRangeMin(wigData[chromIndex].Values, records[k].ChromStart, records[k].ChromEnd)
+				} else if bedAverageWigFlag {
+					currValue = bedRangeAverage(wigData[chromIndex].Values, records[k].ChromStart, records[k].ChromEnd)
+				} else {
+					currValue = bedRangeMax(wigData[chromIndex].Values, records[k].ChromStart, records[k].ChromEnd)
+				}
 				if norm == true {
-					maxWig = maxWig / wigTotal
+					currValue = currValue / wigTotal
 				}
-				currentBed.Annotation = append(currentBed.Annotation, fmt.Sprintf("%g", float64(maxWig))) // %g will
+				currentBed.Annotation = append(currentBed.Annotation, fmt.Sprintf("%g", currValue)) // %g will
```

Comment:
Will it be beneficial to mark in some way whether we calculated max, min or average, either in the name field in each entry, or somewhere in the output file name? It might be annoying to have to remove something in each name field if we don't want it to contain a "max/min/avg" label, but I wonder if indicating "_min/max/avg" in the output file name will be useful